libs/sdk/android/src/main/java/com/rudderstack/react/android/Utility.java

```diff
@@ -126,12 +130,23 @@ static RudderOption convertReadableMapToOptions(ReadableMap readableMap) {
                     }
                 }
             }
+            setCustomContext(optionsMap, options);
         } catch (Exception e) {
             RudderLogger.logError("Exception occurred while handling options: " + e.getMessage());
         }
         return options;
     }
 
+    private static void setCustomContext(Map<String, Object> optionsMap, RudderOption options) {
+        for (String key : optionsMap.keySet()) {
+            if (!(key.equals(EXTERNAL_ID) || key.equals(EXTERNAL_IDS) || key.equals(INTEGRATIONS))){
+                if (optionsMap.get(key) instanceof Map && !isEmpty(optionsMap.get(key))) {
```

Comment:
NOTE: In native Android and iOS, we don't support primitive context.

Do you meant like this (someStringKey1 and someStringKey2):
```
{
  tier: {
      category: 'premium',
      type: 'gold',
  },
  someStringKey1: 'someStringValue1`,
  someStringKey2: 123,
}
```

Actually, the [Android](https://github.com/rudderlabs/rudder-sdk-android/blob/7176a6851d5782ff15489a219f415e7236e8e086/core/src/main/java/com/rudderstack/android/sdk/core/RudderOption.java#L67) and [iOS](https://github.com/rudderlabs/rudder-sdk-ios/blob/8f102dac469826c626b7d43edb6ac09b0e4d794a/Sources/Classes/RSOption.m#L77) `putCustomContext` APIs expect the value to be of type `Map/Dictionary`. 

Please let me know if we should still support primitive types (in the customContext) and also please guide me with the steps to convert the value to map/dictionary types. 

If you meant something else then please let me know.

Reply:
Yes, I don't see a reason why we should not support contextual values that are not objects.

Even here, the input `options` is an object out of which we're excluding `integrations` and `extenalId(s)` fields, which will eventually give us an object with fields of different values.

Is there any blocker with the implementation?

Reply:
Yes, if we try to support the primitive values in RN, it'll require changes (basically introducing a new API) in the core Android and iOS SDKs. Before that we cannot support these changes in RN.

Currently, there is no way to set the primitive types value in Android and iOS and it will be a new feature request

Please let me know how should we proceed further?